src/storm/models/sparse/Model.cpp

```diff
@@ -453,7 +455,98 @@ namespace storm {
                     outStream << "}" << std::endl;
                 }
             }
+            
+            template<typename ValueType, typename RewardModelType>
+            void Model<ValueType, RewardModelType>::writeJsonToStream(std::ostream& outStream) const {
+                using JsonValueType = storm::RationalNumber;
+                storm::json<JsonValueType> output;
+                for (uint64_t state = 0; state < getNumberOfStates(); ++state) {
+                    storm::json<JsonValueType> stateChoicesJson;
+                    stateChoicesJson["id"] = state;
+                    if (hasStateValuations()) {
+                        stateChoicesJson["s"] = getStateValuations().template toJson<JsonValueType>(state);
+                    }
+                    auto labels = getLabelsOfState(state);
+                    if (!labels.empty()) {
+                        stateChoicesJson["lab"] = labels;
```

Comment:
I would not make lab optional, but rather put in an empty array if there are no labels.

Reply:
Agreed.